client/src/driver/BitbakeRecipeScanner.ts

```diff
@@ -66,7 +68,11 @@ export class BitbakeRecipeScanner {
             if (this.scanResults !== '' && uri !== undefined && chosenRecipe !== undefined) {
               logger.debug('[onDidEndTask] Sending recipe environment to the server')
               const requestParam: RequestParams['ProcessRecipeScanResults'] = { scanResults: this.scanResults, uri, chosenRecipe }
-              await this._languageClient.sendNotification(RequestMethod.ProcessRecipeScanResults, requestParam)
+              const tasks = await this._languageClient.sendRequest<RequestResult['ProcessRecipeScanResults']>(RequestMethod.ProcessRecipeScanResults, requestParam)
+              if (tasks !== undefined) {
+                logger.debug(`Request <ProcessRecipeScanResults> returned, Update available tasks for the recipe ${chosenRecipe}`)
+                this.updateTasksForRecipe(chosenRecipe, tasks)
```

Comment:
I implemented a generic way to get variable values from the server to the client, with the option to launch the scan immediatly or not. It's more generic than storing them one by one each scan. Also it can trigger a synchronous recipe scan (which is not needed here). PR: https://github.com/yoctoproject/vscode-bitbake/pull/115

I think we should rather go with this generic approach as we use more and more variables.

Reply:
Re-implement the feature using the getVariableValue function